keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/ArtifactSummary.kt

```diff
@@ -58,12 +58,14 @@ data class ArtifactSummaryInEnvironment(
   val replacedAt: Instant? = null,
   val replacedBy: String? = null,
   val isPinned: Boolean = false,
-  val pinned: Pinned? = null,
+  val pinned: ActionMetadata? = null,
+  val isVetoed: Boolean = false,
```

Comment:
Is it ever possible for `isVetoed` to be true and `vetoed` to be null? If not, I would just change `isVetoed` to a property that checks if `vetoed` is null. (Same with `pinned`)

Reply:
That's actually what we do when we create this class. I think you're right, it would be better to do it as a property